cmd/kube-proxy/app/server_test.go

```diff
@@ -501,6 +501,10 @@ udpIdleTimeout: 250ms`)
 		errCh := make(chan error, 1)
 		go func() {
 			errCh <- opt.runLoop()
+			select {
```

Comment:
Can you please give an example when it failed?
I am just curious how adding the select should fix it?

Reply:
I used a debugger to run this unit test, and found that (*Options).eventHandler() is called once, by this call chain:
err = opt.Complete() at cmd/kube-proxy/app/server_test.go:495
if err := o.initWatcher(); err != nil { at cmd/kube-proxy/app/server.go:233
err := fswatcher.Init(o.eventHandler, o.errorHandler) at cmd/kube-proxy/app/server.go:248
w.watcher, err = fsnotify.NewWatcher() at pkg/util/filesystem/watcher.go:63
go w.readEvents() at vendor/github.com/fsnotify/fsnotify/inotify.go:59
case w.Events <- event: at vendor/github.com/fsnotify/fsnotify/inotify.go:285

Reply:
I think eventHandle() is called only for the first unit test. Sometimes, it is called once, but sometimes it is called twice. The call chain is exactly the same for the two runs.